src/commands/verify_url_links_command.rs

```diff
@@ -0,0 +1,86 @@
+use super::command_interface::Command;
+use reqwest;
+use futures::future::join_all;
+
+pub struct VerifyUrlLinksCommand {
+    urls: Vec<String>,
+}
+
+#[derive(Debug)]
+pub struct UrlLinksResult {
+    pub valid_urls: Vec<String>,
+    pub invalid_urls: Vec<String>,
+}
+
+impl VerifyUrlLinksCommand {
+    pub fn new(urls: Vec<String>) -> Self {
+        VerifyUrlLinksCommand { urls }
+    }
+}
+
+impl Command for VerifyUrlLinksCommand {
+    type Output = UrlLinksResult;
+
+
+    fn execute(&self) -> UrlLinksResult {
+        let mut valid_urls = Vec::new();
+        let mut invalid_urls = Vec::new();
+
+        async fn check_one_url(url: &str) -> (String, Result<reqwest::Response, reqwest::Error>) {
```

Comment:
That return type looks to me slightly bizarre

Reply:
Also, is there a special need to use nested functions?

Reply:
> Also, is there a special need to use nested functions?

Reason: encapsulation)